design/accepted/gc_plus/gc-plus-cloud.md

```diff
@@ -0,0 +1,52 @@
+# Managed Uncommitted Garbage Collection for lakeFS cloud - Execution Plan
+
+## Goals
+1. Run UGC for lakeFS cloud users
+2. Allow Cloud users to clean up storage space by deleting objects from the underlying storage for cost efficiency
+3. Support new and existing cloud users (both repository structures)
+
+## Non-Goals
+1. Remove lakeFS limitations during the UGC job run
+2. Improve UGC OSS support
+
+### User Story
+When creating a new cloud installation - on the GC initializations page,
+add UGC initialization (default run job=true for all repositories, default sweep = true).
+It can get the same rules as the GC.
+Add a link to the documentation costrains (lakeFS writes during the job, explain debugging and report).
+
+### Permissions
+Create default UGC user with permissions for every new cloud env (The permissions to Enigma was added to the GC user)
+- lakeFS permissions - prepare uncommitted gc
+- Bucket permissions - role to list, write, delete
+
+### Managed UGC job
+- Control plane interface (design with cloud native team)
+  - Add step for the create cloud installation
+  - UGC to get configuration for the spark job (configured role, secret and access key, installation endpoint, repository)
+  - User to configure min age seconds time
```

Comment:
Why should we let them configure this? It requires deep understanding of the system.

Reply:
maybe they want to keep the data for X hours before deleting it in case of wanting to revert

Reply:
AFAIK there's no revert operation for uncommitted data